packages/components/src/autocomplete/autocompleter-ui.tsx

```diff
@@ -129,7 +129,7 @@ export function getAutoCompleterUI( autocompleter: WPCompleter ) {
 		} ) => (
 			<Component
 				id={ listBoxId }
-				role="listbox"
+				role="combobox"
```

Comment:
Not sure how valid this is but it seems to actually work really well. Still need to test on Mac but it eliminates the re-rendering feeling. I did inspect with React Dev Tools and while there are some passed properties that do force some re-rendering, I think this issue was more related to the listbox role itself.

Reply:
Changing this to `combobox` doesn't seem to make any difference on Mac, I guess because VoiceOver has no idea what it's doing anyway! Either way, it's definitely much more vocal now, announcing each option as it's selected, and debouncing appropriately.

Academically speaking, I suppose it should be the controlling "paragraph" that temporarily gets a role change, while the popup should stay as a `listbox`. When I hacked that together very quickly, VoiceOver announced the following...

> You are currently on a Menu pop-up combo box, inside a frame. Type text or, to display a list of choices, press Control-Option-Space.

That feels somehow more "correct", in terms of what it is semantically, though confusingly pressing <kbd>Control</kbd>+<kbd>Option</kbd>+<kbd>Space</kbd> doesn't do anything. Project for another time!

Reply:
@andrewhayward If I keep the `role="listbox"`, every time I press Up or Down Arrow, it has this re-render effect. I think its because the passed selected index prop changes.

Voiceover, now that I think about it, won't work at all even with this change due to Apple not supporting `aria-owns`.

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-owns

> Warning: At the time of this writing, aria-owns is not supported on MacOS and iOS with VoiceOver.

I might try switching with `aria-controls`.

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-controls

You should look at how we implemented the post author combo box if there are more than 25 users on the site. You can find that in the editor package, I think it uses `Combobox` component which wraps around part of `FormTokenInput`.

Reply:
Technically speaking, this change shouldn't affect how React re-renders items. As you said, there's probably a different change that triggers re-render.

Reply:
@andrewhayward Okay, after doing some more debugging, we could improve a little more. It would be possible to keep `role="combobox"` and `role="listbox"` on the correct elements if we could prevent `listBox` from re-rendering every time. Every time I press Up or Down Arrow keys, the `AutocompleterUI` triggers a re-render of the `ListBox` and React Dev Tools says the cause is:

`Props changed: (selectedIndex, onChangeOptions, onSelect, reset)`

Is it possible to `useCallback` or `useMemo` the `onChangeOptions` callback to only change if needed? Up or Down Arrow keys should only adjust the `selectedIndex`, not re-render the entire options tree.

Reply:
From what I can tell, the re-render is being caused by the changing props returned on the hook. Not sure how to go about fixing this so the suggestions list only re-renders when the filter is used. This doesn't appear to be a problem in the `Combobox` component. In theory, the props would simply update the existing component but it seems like the whole component is being replaced which also forces `ListBox` to re-render.